monitoring/src/main/java/org/polypheny/db/monitoring/events/analyzer/QueryEventAnalyzer.java

```
@@ -0,0 +1,97 @@
+                .executionTime( queryEvent.getExecutionTime() )
+                .rowCount( queryEvent.getRowCount() )
+                .isSubQuery( queryEvent.isSubQuery() )
+                .recordedTimestamp( queryEvent.getRecordedTimestamp() )
+                .build();
+
+        RelRoot relRoot = queryEvent.getRouted();
+        if ( relRoot != null ) {
+            RelNode node = relRoot.rel;
+            processRelNode( node, queryEvent, metric );
+        }
+
+        if ( queryEvent.isAnalyze() ) {
+            processDurationInfo( queryEvent, metric );
+
+        }
+
+        return metric;
+    }
+
+
+    private static void processDurationInfo( QueryEvent queryEvent, QueryDataPoint metric ) {
+        try {
+            InformationDuration duration = new Gson().fromJson( queryEvent.getDurations(), InformationDuration.class );
+            getDurationInfo( metric, "Plan Caching", duration );
+            getDurationInfo( metric, "Index Lookup Rewrite", duration );
+            getDurationInfo( metric, "Constraint Enforcement", duration );
+            getDurationInfo( metric, "Implementation Caching", duration );
+            getDurationInfo( metric, "Index Update", duration );
+            getDurationInfo( metric, "Routing", duration );
+            getDurationInfo( metric, "Planning & Optimization", duration );
+            getDurationInfo( metric, "Implementation", duration );
+            getDurationInfo( metric, "Locking", duration );
+        } catch ( Exception e ) {
+            log.debug( "could not deserialize of get duration info" );
+        }
+    }
+
+
+    private static void getDurationInfo( QueryDataPoint queryMetric, String durationName, InformationDuration duration ) {
+        try {
+            long time = duration.getDuration( durationName );
+            queryMetric.getDataElements().put( durationName, time );
+        } catch ( Exception e ) {
+            log.debug( "could no find duration:" + durationName );
```

This more or less ignores the exception...